parser/src/powdr.lalrpop

```diff
@@ -684,6 +694,44 @@ PatternList: Vec<Pattern> = {
     => vec![]
 }
 
+// ---------------------------- Trait/Impl Declaration -----------------------------
+
+TraitDeclaration: TraitDeclaration<Expression> = {
+    "trait" <name:Identifier> <type_vars:("<" <TypeVarBounds> ">")?> "{" <methods:TraitMethods> "}" => TraitDeclaration { name, type_vars: type_vars.unwrap_or_default(), methods }
```

Comment:
I'm wondering, maybe it is easier to not allow restricting the type vars in the trait declaration but only in the implementation?